### PR TITLE
CASSANDRA-18921-4.1: Fix DescribeStatement

### PR DESCRIPTION
**Pull request by jacek-lewandowski**

### CODE REVIEW

File: src/java/org/apache/cassandra/cql3/statements/DescribeStatement.java [hunk 1]
```diff
@@ -132,8 +132,13 @@ public final ResultMessage execute(QueryState state, QueryOptions options, long
...
     public ResultMessage executeLocally(QueryState state, QueryOptions options)
     {
-        Keyspaces keyspaces = Schema.instance.distributedAndLocalKeyspaces();
-        UUID schemaVersion = Schema.instance.getVersion();
+        UUID schemaVersion;
+        Keyspaces keyspaces;
+        do
+        {
+            schemaVersion = Schema.instance.getVersionSync();
+            keyspaces = Schema.instance.distributedAndLocalKeyspaces();
```
Comment (smiklosovic) [inline src/java/org/apache/cassandra/cql3/statements/DescribeStatement.java:140]: @jacek-lewandowski this seems interesting at first sight ... feels unnatural. Basically, we set `keyspaces` but they are not used in `while`. Is that `distributedAndLocalKeyspaces` doing something which will make `getVersionSync` behave differently? Just asking ...

Comment (smiklosovic) [reply src/java/org/apache/cassandra/cql3/statements/DescribeStatement.java:140]: Ah never mind ... I was reading this change wrongly. `keyspaces` are used after this do-while ... Forget it.

File: src/java/org/apache/cassandra/schema/Schema.java [hunk 2]
```diff
@@ -496,25 +496,43 @@ public Optional<Function> findFunction(FunctionName name, List<AbstractType<?>>
...
+        return version;
+    }
+
     /**
      * Checks whether the given schema version is the same as the current local schema.
+     * <p>
+     * @deprecated please use {@link #getVersionSync()} or {@link #getVersion()} depending on the use case
      */
-    public boolean isSameVersion(UUID schemaVersion)
+    @Deprecated(since = "4.1", forRemoval = true)
```
Comment (smiklosovic) [inline src/java/org/apache/cassandra/schema/Schema.java]: :1st_place_medal:

**Comment by jacek-lewandowski** [comment]
https://app.circleci.com/pipelines/github/jacek-lewandowski/cassandra/1040/workflows/a44e95b0-67a5-4e89-88f6-2680dedd80e0 🆗

**Comment by jacek-lewandowski** [comment]
https://app.circleci.com/pipelines/github/jacek-lewandowski/cassandra/1040/workflows/1020c55f-f59e-4ac6-ab0a-f589bbc1050a (j11) 🆗

**Pull request closed without merging**